### ISSUE TITLE
Disable NVIDIA Management Library (NVML) in Intel mode

### ISSUE THREAD

**Issue by zli117** [Askannz/optimus-manager#66]
I'm having an issue that when you switch to Intel mode, Nouveau wouldn't power off GPU even when there's no load on it, and even reboot couldn't fix. When checking `dmesg` (in Intel mode), I noticed these messages happen around every three seconds:
```
[ 1415.064458] nvidia-nvlink: Nvlink Core is being initialized, major device number 236
[ 1415.064692] NVRM: The NVIDIA probe routine was not called for 1 device(s).
[ 1415.064693] NVRM: This can occur when a driver such as: 
               NVRM: nouveau, rivafb, nvidiafb or rivatv 
               NVRM: was loaded and obtained ownership of the NVIDIA device(s).
[ 1415.064693] NVRM: Try unloading the conflicting kernel module (and/or
               NVRM: reconfigure your kernel without the conflicting
               NVRM: driver(s)), then try loading the NVIDIA kernel module
               NVRM: again.
[ 1415.064693] NVRM: No NVIDIA graphics adapter probed!
[ 1415.107444] nvidia-nvlink: Unregistered the Nvlink Core, major device number 236
```
which seems to me that some background service is constantly trying to restart Nvlink Core. Then I found after removing the NVML library file `/usr/lib/libnvidia-ml.so.418.56` from `LD_LIBRARY_PATH`, the behavior will stop and GPU is properly powered off. 

I'm wondering whether it's possible to disable the NVML library when switching to Intel and reenable it once switching back, as `nvidia-smi` won't be functional either way in Intel mode? Thanks in advance!

System info:
 * Hardware: Lenovo Thinkpad P51 NVIDIA Quadro M2200 4GB
 * OS: Arch Linux 5.0.7-arch1-1-ARCH
 * Desktop environment: KDE 5.15.4
 * Display manager: SDDM 0.18.1-1

**Comment by Askannz** [comment]
Sorry for the late reply, and thanks for bringing that up.

Could you test the latest Git version and see if the problem persists ? For this new version I more or less copied the method Ubuntu uses, and they don't seem to have that problem, so there is a chance it could fix your issue. If that doesn't work, we'll try to find what is loading this library and how to stop it.

The new version introduces a lot of changes to the way optimus-manager works, so you need to reboot after upgrading to make sure everything is reloaded up properly.

**Comment by zli117** [comment]
Sure, will try on a fresh install after my finals. Thanks for the reply!

**Comment by zli117** [comment]
Hi Askannz, I think I've figured out why. First, I found every time I run nvidia-smi in intel mode will for sure bring GPU into active state, and removing NVML library will prevent this from happening. However, after several seconds, GPU will power back off, which is normal. However, I installed a thermal monitor widget on KDE, and tracked GPU temperature using readings from nvidia-smi. As a result, the widget constantly retries nvidia-smi, preventing GPU from powering off. After disabling the tracking, GPU is properly powered off when switched into intel mode.
